File: spring/blog/src/main/java/development/blog/data/PostData.java
package development.blog.data;

import development.blog.logic.User;
import development.blog.representations.response.PostResponse;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

import java.sql.*;
import java.util.ArrayList;

@Service
public class PostData {
    static final Logger log = LoggerFactory.getLogger(User.class);
    CommonData commonData;
    Connection connection;

    public PostData(Database database, CommonData commonData) {
        this.connection = database.getConnection();
        this.commonData = commonData;
    }

    public ArrayList<PostResponse> getAllPosts() throws SQLException {
        final String sql = "SELECT idpost,title FROM public.post;";
        PreparedStatement statement =
                this.connection.prepareStatement(sql);
        ResultSet rs = statement.executeQuery();
        ArrayList<PostResponse> posts = new ArrayList<PostResponse>();

        while (rs.next()) {
            posts.add(new PostResponse(
                    rs.getLong(1),
                    rs.getString(2)
            ));
        }
        return posts;
    }

    public PostResponse getPost(Long id) throws SQLException {
        final String sql = "SELECT idpost,title FROM public.post WHERE idpost=?;";
        PreparedStatement statement =
                this.connection.prepareStatement(sql);
        statement.setLong(1, id);

        ResultSet rs = statement.executeQuery();

        rs.next();
        return new PostResponse(
                rs.getLong(1),
                rs.getString(2)
        );
    }

    public Long addPost(String title) throws SQLException {
        final String sql = "INSERT INTO public.post (title) VALUES (?);";
        PreparedStatement statement =
                this.connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
        statement.setString(1, title);
        statement.executeUpdate();

        ResultSet rs = statement.getGeneratedKeys();
        rs.next();

        return rs.getLong(1);
    }

    public void deletePost(Long id) throws SQLException {
        commonData.deleteById(id, "comment", "post_idpost");
        commonData.deleteById(id, "segment", "post_idpost");
        commonData.deleteById(id, "post", "idpost");
    }
}

File: flask/blog/blog/data/post_data.py
from typing import Any
from data import database


def all_posts() -> list[dict[str, Any]]:
    posts = database.find_all("SELECT idpost,title FROM public.post;", [])

    reshaped = list(map(lambda post: {"idpost": post[0], "title": post[1]}, posts))

    return reshaped


def post_by_id(post_id: int)-> Any:
    return database.find_single(
        "SELECT idpost,title FROM public.post WHERE idpost=%s;", [str(post_id)]
    )


def create_post(post_title: str) -> int:
    return database.insert_data_returning_id(
        "INSERT INTO public.post (title) VALUES (%s) RETURNING idpost;", [post_title]
    )


def delete_post(post_id: int):
    database.delete_all("DELETE FROM public.comment WHERE post_idpost=%s", [str(post_id)])
    database.delete_all("DELETE FROM public.segment WHERE post_idpost=%s", [str(post_id)])
    database.delete_all("DELETE FROM public.post WHERE idpost=%s", [str(post_id)])
